jf_agent/main.py

```diff
@@ -605,7 +372,16 @@ def upload_file(filename, path_to_obj, signed_url):
 
     print('Sending data to Jellyfish... ')
 
-    signed_urls = get_signed_url(os.listdir(config.outdir))
+    # obtain file names from the directory
+    _, directories, filenames = next(os.walk(config.outdir))
```

Comment:
This doesn't look right -- doesn't `next` just get a single item?

Though see my comment above -- maybe we should just write all the files into the same directory and keep the rest of this code more or less unchanged.

Reply:
This is right. `os.walk` returns an iterator, with each item being the next level of subdirectories. (I only want the first)

If I used `listdir()`, it would not show me which were directories, and which were filenames.